### PR TITLE
Adds error handling for Bitcoin.com widget

### PR DESCRIPTION
**Pull request by ryanml**
Fixes brave/brave-browser#11960

Screen Shot 2020-10-08 at 5 43 05 PM

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Defined in issue

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Review by emerick — APPROVED** [review]
LGTM

File: components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx [hunk 1]
```diff
@@ -283,9 +316,14 @@ class BitcoinDotCom extends React.PureComponent<Props, State> {
   }
 
   renderFooter () {
+    const { amountError, currentAmount } = this.state
+
     return (
       <Styled.FooterWrapper>
-        <Styled.BuyButton onClick={this.openBuyURL}>
+        <Styled.BuyButton
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx:323]: nit: `disabled` if there is an error?

File: components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts [hunk 2]
```diff
@@ -226,7 +249,7 @@ export const BuyButton = styled<{}, 'button'>('button')`
   padding: 12px 0px;
   font-weight: bold;
   font-size: 13px;
-  cursor: pointer;
+  cursor: ${p => p.error ? 'default' : 'pointer'};
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts:252]: nit: disabled style

File: components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx [hunk 3]
```diff
@@ -18,6 +18,7 @@ interface State {
   currentAmount: string
   assetsShowing: boolean
   fiatCurrenciesShowing: boolean
+  amountError: boolean
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx:21]: nit: if this were a string then you could support multiple types of error (`null` would mean no error), but I guess here there is only 1 invalid amount error case?

Comment (ryanml) [reply components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx:21]: Correct yeah, this is the only path to user error at the moment

File: components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts (hunk 2 above)
Comment (ryanml) [reply components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts:252]: fixed

File: components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx (hunk 1 above)
Comment (ryanml) [reply components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx]: done

File: components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts [hunk 4]
```diff
@@ -149,6 +150,21 @@ export const AmountInput = styled<StyleProps, 'input'>('input')`
...
+    ` : ''
+  }
+
+  ${(p) => p.error
+    ? css`
+      &:focus {
+        outline: none;
+      }
+    ` : ''
+  }
```
Comment (AlanBreck) [inline components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts:167]: If I'm not mistaken, the above can be simplified to something like:

```js
${(p) => p.error && `
    border: 1px solid #ff0000;
    border-right: none;

    &:focus {
      outline: none;
    }`
}
```

File: components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts [hunk 5]
```diff
@@ -176,6 +192,13 @@ export const FiatDropdown = styled<StyleProps, 'div'>('div')`
       border-bottom-right-radius: 0px;
     ` : ''
   }
+
+  ${(p) => p.error
+    ? css`
+      border: 1px solid #ff0000;
+      border-left: 1px solid lightgray;
+    ` : ''
+  }
```
Comment (AlanBreck) [inline components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts:201]: Like above. I think a short circuit condition would be appropriate:

```
${(p) => p.error && `
      border: 1px solid #ff0000;
      border-left: 1px solid lightgray;
    `
  }
```

File: components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts [hunk 6]
```diff
@@ -227,6 +250,10 @@ export const BuyButton = styled<{}, 'button'>('button')`
   font-weight: bold;
   font-size: 13px;
   cursor: pointer;
+
+  &:disabled {
+    pointer-events: none;
```
Comment (AlanBreck) [inline components/brave_new_tab_ui/components/default/bitcoinDotCom/style.ts:255]: Possibly drop the opacity when disabled?

**Review by AlanBreck — COMMENTED** [review]
LGTM. Left a few minor comments.

**Review by AlanBreck — APPROVED** [review]

**Pull request merged**